### ISSUE TITLE
[Python][P1] ml/marl: MAPPO softmax overflow produces NaN / argmax=0 fleet actions

### ISSUE THREAD

**Issue by ionfwsrijan** [KanishJebaMathewM/Truxify#11388]
## Description
`backend/ml/marl/mappo_fleet.py` computes the stochastic policy via a numerically unstable softmax:

```python
# mappo_fleet.py:23-27
logits = np.dot(state, self.actor_weights)   # can be large
probs = np.exp(logits) / np.sum(np.exp(logits))
action = np.argmax(probs)
```

## Actual Behavior
`action_logits` is an unbounded dot product of state×weights, so `np.exp(logits)` overflows to `inf` for any moderately large logit. The `inf/inf` (or `inf/sum`) then yields `NaN`, and `np.argmax(NaN)` returns `0`, so the fleet agent collapses to a fixed action regardless of state. Large/negative logits conversely underflow to `0`, distorting the distribution.

## Expected Behavior
A numerically stable softmax (subtract the max before exponentiating) so probabilities are well-defined and `argmax` reflects the actual policy.

## Proposed Fix
```python
z = logits - np.max(logits)          # stability shift
e = np.exp(z)
probs = e / np.sum(e)
```
Add a guard for the degenerate all-`-inf` case and a unit test asserting finite `probs` and `action` for large-magnitude logits. This changes the MARL fleet re-balancing decisions and the surrounding inference path.
